### PR TITLE
ZEPPELIN-1866. No meaningful exception for pig interpreter under tez engine

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
This PR is to fix the bug that there's no meaningful exception when using tez engine. 

### What type of PR is it?
[Bug Fix | Improvement ]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-1866

### How should this be tested?
Unit test is added and also manually test it. 

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@felixcheung Please help review. CI failure is not relevant.
```
AngularElem
- should provide onclick method *** FAILED ***
  The code passed to eventually never returned normally. Attempted 1 times over 185.23286 milliseconds. Last failure message: 0 was not equal to 1. (AbstractAngularElemTest.scala:64)
AngularElem
```

File: pig/src/main/java/org/apache/zeppelin/pig/PigUtils.java [hunk 1]
```diff
@@ -171,6 +171,15 @@ public static String extractFromSimplePigStats(SimplePigStats stats) {
   private static String extractFromTezPigStats(TezPigScriptStats stats) {
 
     try {
+      if (stats.getReturnCode() == PigRunner.ReturnCode.UNKNOWN) {
+        LOGGER.warn("unknown return code, can't display the results");
```
Comment (felixcheung) [inline pig/src/main/java/org/apache/zeppelin/pig/PigUtils.java]: should we log that return code then?

Comment (zjffdu) [inline pig/src/main/java/org/apache/zeppelin/pig/PigUtils.java]: It helps for diagnosing.

Comment (felixcheung) [inline pig/src/main/java/org/apache/zeppelin/pig/PigUtils.java]: I reread this, I guess if `getReturnCode` is returning UNKNOWN, there isn't any other meaningful code to put into log. So I'm good with the LOGGER.warn line.

**Comment by felixcheung** [comment]
the same test profile failed in the last 2 CI runs, is that something to be concerned about?

**Comment by zjffdu** [comment]
I think these are flaky test. Because all the failed test are not pig related. Changes in this PR is only on pig interpreter.

```
Failed tests: 
  NotebookTest.testAbortParagraphStatusOnInterpreterRestart:760 expected:<ABORT> but was:<RUNNING>
 
Tests run: 148, Failures: 1, Errors: 0, Skipped: 0
```

```
Results :
 
Tests in error: 
  ZeppelinIT.testAngularRunParagraph:315->AbstractZeppelinIT.waitForText:84 Â» StaleElementReference
 
```

**Comment by felixcheung** [comment]
Cool. It also looks like CI is passing.
Merging if no more comment.

**Comment by felixcheung** [comment]
Cool. ~~It also looks like CI is passing.~~ Hmm, I guess not..
Merging if no more comment.

**Pull request closed without merging**